### DISCUSSION TITLE
Tray Icon does not appear on Raspberry Pi

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#15360] [Q&A]
For some reasons, my tray icon does not appear on my Raspberry pi, works fine on my mint install.

![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/50aea050-4c38-49f3-ab22-53a921a265ea)

![Tray Icon in Mint](https://i.imgur.com/me99tbC.png)

I have checked that the app is running and no exceptions occur, even after use of RPC.

**Comment by timunie** [comment]
I believe there was a recent PR for that. Either it's still not merged and able to test or you can try nightly.

**Reply by Mrcubix** [reply]
Will be trying it out then and let you know

**Reply by Mrcubix** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/d5210242-dba4-45c2-b003-8f6f62b6f9d7)

Still does not show up, No exceptions still.

**Reply by workgroupengineering** [reply]
Try with dbus spy tool for debug issue.

**Reply by Mrcubix** [reply]
will need links as there are a couple repos & tools with similar but not exact name.

Will also need steps on using these tools & get the data required.

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/Avalonia/pull/15298

**Reply by Mrcubix** [reply]
Does not change anything, still does not work, no icons.

On a side note, the PR from that CI build suffer from an obvious bug, that prevent compilation if debug symbols are set to embedded on release, as it will look for a pdb file that does not exist in the current situation.

```log
11.2.999-cibuild0047112-alpha\buildTransitive\AvaloniaBuildTasks.targets(137,5): error MSB4018: 
The "CompileAvaloniaXamlTask" task failed unexpectedly.

11.2.999-cibuild0047112-alpha\buildTransitive\AvaloniaBuildTasks.targets(137,5): error MSB4018: 
System.IO.FileNotFoundException: Could not find file 'WirelessParser\WirelessKitAddon.UX.Desktop\obj\Release\net8.0\linux-arm64\WirelessKitBatteryStatus.UX.pdb'.

11.2.999-cibuild0047112-alpha\buildTransitive\AvaloniaBuildTasks.targets(137,5): error MSB4018: File name: 'WirelessParser\WirelessKitAddon.UX.Desktop\obj\Release\net8.0\linux-arm64\WirelessKitBatteryStatus.UX.pdb'
  at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite) 
  at Avalonia.Build.Tasks.CompileAvaloniaXamlTask.CopyAndTouch(String source, String destination) in /_/src/Avalonia.Build.Tasks/CompileAvaloniaXamlTask.cs:line 54
  at Avalonia.Build.Tasks.CompileAvaloniaXamlTask.Execute() in /_/src/Avalonia.Build.Tasks/CompileAvaloniaXamlTask.cs:line 41   
  at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() 
  at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)

[WirelessParser\WirelessKitAddon.UX.Desktop\WirelessKitAddon.UX.Desktop.csproj]
```

**Comment by maxkatz6** [comment]
If one distro shows tray icon, but another doesn't, there is a big chance, that that another doesn't support this DBus protocol at all.

Typically, you can debug it with d-feet tool and see what protocols distro implements. 
Avalonia uses `org.kde.StatusNotifierItem` protocol. It seems there is less widely used `org.freedesktop.StatusNotifierItem` now, which we could support as well.

**Reply by Mrcubix** [reply]
I see no KDE elements at all, but i do see freedesktop protocols, such as Notifications which i'm using for this application.
It doesn't seem like StatusNotifierItem is currently active, meaning RealVNC makes use of another method to show their tray icon.
